### PR TITLE
docs(dbt): add deprecated markers in `dagster-dbt` API docs

### PR DESCRIPTION
**Pull request by rexledesma**
## Summary & Motivation
We were using `deprecation_warning`, which displays a warning at runtime. Use `deprecated` instead to also display that the API is deprecated in our API docs.

`deprecated` doesn't work on `OpDefinition`. Following a pattern like `is_resource_def` and adding `is_op_def` at https://github.com/dagster-io/dagster/blob/5d989457b7a7aa2d6fb511228cbd748c2193d29b/python_modules/dagster/dagster/_core/decorator_utils.py#L159 didn't work out of the box, so skip marking op defs, and continue using `deprecation_warning`.

## How I Tested These Changes
See docs

### CODE REVIEW

**Comment by rexledesma** [comment]
Current dependencies on/for this PR:
* `master`
  * **PR #19140** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/19140?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/19140?utm_source=stack-comment).

File: python_modules/libraries/dagster-dbt/dagster_dbt/__init__.py [hunk 1]
```diff
@@ -174,19 +153,59 @@
...
+    #     "dagster.some.module",
+    #     "1.1.0",  # breaking version
+    #     "Use Bar instead.",
+    # ),
+    **{
+        value: (
+            module,
+            "0.24.0",
+            (
+                "Use the @op decorator and DbtCliResource instead.\n\n"
```
Comment (erinkcochran87) [inline python_modules/libraries/dagster-dbt/dagster_dbt/__init__.py:177]: Does this warning also display in logs? If not, could we clean up some of the formatting so it's more readable in the docs? It's a pretty big blob at the moment:

Screenshot 2024-01-10 at 12 26 21 PM

Comment (rexledesma) [reply python_modules/libraries/dagster-dbt/dagster_dbt/__init__.py:177]: https://github.com/dagster-io/dagster/pull/19151 should resolve

**Review by erinkcochran87 — APPROVED** [review]
ty! I'd like Sean to look at the other PR you mentioned, but I'm good with this.

**Pull request merged**